java/client/src/org/openqa/selenium/remote/internal/OkHttpClient.java

```diff
@@ -116,11 +117,20 @@ public void close() {
 
     @Override
     public HttpClient createClient(URL url) {
-      okhttp3.OkHttpClient client = new okhttp3.OkHttpClient.Builder()
+      okhttp3.OkHttpClient.Builder builder = new okhttp3.OkHttpClient.Builder()
           .connectionPool(pool)
           .followRedirects(true)
-          .followSslRedirects(true)
-          .build();
+          .followSslRedirects(true);
+
+      String urlUserInfo = url.getUserInfo();
+      if (urlUserInfo != null) {
+        UserInfo userInfo = new UserInfo(urlUserInfo);
+        String basic = Credentials.basic(userInfo.getUser(), userInfo.getPassword());
+        builder.addInterceptor(chain ->
+          chain.proceed(chain.request().newBuilder().header("Authorization", basic).build())
```

Comment:
This will loop endlessly if requests aren't authorized properly, according to the okhttp docs.

Reply:
it's not so, this code adds just an interceptor, not `Authenticator`, because it allows to avoid an extra HTTP call, please check https://stackoverflow.com/a/36056355/2067574 for more details